src/main/java/seedu/duke/ExpenseTracker.java

```diff
@@ -0,0 +1,320 @@
+package seedu.duke;
+
+import java.text.DecimalFormat;
+import java.math.BigDecimal;
+import java.math.RoundingMode;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Calendar;
+
+public class ExpenseTracker {
+    private List<Category> categories;
+    private List<Expense> expenses;
+    private Map<Category, Budget> budgets = new HashMap<>();
+    private boolean isautoResetEnabled;
+    private int lastResetMonth;
+
```

Comment:
Consider separating the class for better readability and maintainability. Having different classes for Expense, Category, and Budget would be better.